### ISSUE TITLE
Chore: Cleanup Code and prepare for v3

### ISSUE THREAD

**Issue by maennchen** [quantum-elixir/quantum-core#355]
* [x] Make sure that storage API is stable
* [x] Remove dirty code only to prevent breaking changes
* [x] Improve Code Quality

**Comment by jung-hunsoo** [comment]
How about support defining jobs in `Scheduler` module, instead of `config`?

It will help maintaining config clean, and keep codes organized.

**Comment by maennchen** [comment]
@jung-hunsoo This is already possible. I'm not sure though if this is properly documented.

You can define an `init/1` function in the Scheduler where it is possible to alter the configuration in code.

**Comment by jung-hunsoo** [comment]
@maennchen Thanks. Based on your comment, I successfully moved defining jobs from `config` to `Scheduler` module.

Here is my working codes:
```elixir
  def init(config) do
    jobs = [
      # Everyday midnight in Seoul.
      {"5 15 * * *", {Nohogu.Tips, :clear_old_applied_editors, []}},
      # Monthly midnight in Seoul.
      {"10 15 1 * *", {Nohogu.Tips, :rerole_editors_by_role_all, []}}
    ]

    base = Quantum.Job.new(config)
    norm_jobs = Enum.map(jobs, &Quantum.Normalizer.normalize(base, &1))
    Keyword.put(config, :jobs, norm_jobs)
  end
```

Any improvements?

**Comment by maennchen** [comment]
Fixed on master, will be released with #427
